code/mission/missionlog.cpp

```diff
@@ -359,15 +226,15 @@ void mission_log_add_entry(LogType type, const char *pname, const char *sname, i
 
 	// if in multiplayer and I am the master, send this log entry to everyone
 	if ( MULTIPLAYER_MASTER ){
-		send_mission_log_packet( &log_entries[last_entry] );
+		send_mission_log_packet( entry );
 	}
 
-	last_entry++;
-
 #ifndef NDEBUG
-	if ( !(last_entry % 10) ) {
-		if ( (last_entry > LOG_HALFWAY_REPORT_NUM) && (last_entry > last_entry_save) ){
-			nprintf(("missionlog", "new highwater point reached for mission log (%d entries).\n", last_entry));
+	if ( (Log_entries.size() % 10) == 0 ) {
+		static size_t largest_size_save = Log_entries.size();
+		if ( (Log_entries.size() > 350) && (Log_entries.size() > largest_size_save) ){
+			largest_size_save = Log_entries.size();
+			nprintf(("missionlog", "new highwater point reached for mission log (" SIZE_T_ARG " entries).\n", largest_size_save));
 		}
 	}
```

Comment:
This seems kinda superflous now.
I bet this was only there to monitor that a safe size for the missionlog was picked.
I'd honestly advocate to just get rid of this wholesale.
If not, I'd at least want a slight tidyup in moving the minimum count to print about this into the static variable, and not in the if clause like this:
```suggestion
	if ( (Log_entries.size() % 10) == 0 ) {
		static size_t largest_size_save = 350;
		if ( Log_entries.size() > largest_size_save ){
			largest_size_save = Log_entries.size();
			nprintf(("missionlog", "new highwater point reached for mission log (" SIZE_T_ARG " entries).\n", largest_size_save));
		}
	}
```

Reply:
Yeah, fair point, I'll just delete it.